service/cluster/xtra_db_cluster.go

```diff
@@ -80,38 +76,62 @@ func (s *XtraDBClusterService) ListXtraDBClusters(ctx context.Context, req *cont
 			Pxc: &controllerv1beta1.XtraDBClusterParams_PXC{
 				DiskSize: int64(pxcDiskSize),
 			},
-			Proxysql: &controllerv1beta1.XtraDBClusterParams_ProxySQL{
-				DiskSize: int64(proxySQLDiskSize),
-			},
 		}
-		if cluster.PXC.ComputeResources != nil {
-			cpuMillis, err := convertors.StrToMilliCPU(cluster.PXC.ComputeResources.CPUM)
+
+		if cluster.ProxySQL != nil {
+			proxySQLDiskSize, err := convertors.StrToBytes(cluster.ProxySQL.DiskSize)
```

Comment:
this code is quite hard to read, even if we can't solve the nested nature of this bit the least we can do to improve readability is move all the code under the if condition to a separate functions for e.g. `setProxySQLParams` or `setHAProxyParams`

Reply:
Done

Reply:
I ended up refactoring them out. But the `if` blocks are now shorter.